Makefile.pre.in

```diff
@@ -652,28 +652,16 @@ profile-run-stamp:
 	# enabled.
 	$(MAKE) profile-gen-stamp
 	# Next, run the profile task to generate the profile information.
-	$(MAKE) run_profile_task
-	$(MAKE) build_all_merge_profile
+	@ # FIXME: can't run for a cross build
+	$(LLVM_PROF_FILE) $(RUNSHARED) ./$(BUILDPYTHON) $(PROFILE_TASK) || true
+	$(LLVM_PROF_MERGER)
 	# Remove profile generation binary since we are done with it.
 	$(MAKE) clean-retain-profile
 	# This is an expensive target to build and it does not have proper
 	# makefile dependency information.  So, we create a "stamp" file
 	# to record its completion and avoid re-running it.
 	touch $@
 
-.PHONY: build_all_generate_profile
-build_all_generate_profile:
-	$(MAKE) @DEF_MAKE_RULE@ CFLAGS_NODIST="$(CFLAGS_NODIST) $(PGO_PROF_GEN_FLAG)" LDFLAGS_NODIST="$(LDFLAGS_NODIST) $(PGO_PROF_GEN_FLAG)" LIBS="$(LIBS)"
-
-.PHONY: run_profile_task
-run_profile_task:
-	@ # FIXME: can't run for a cross build
-	$(LLVM_PROF_FILE) $(RUNSHARED) ./$(BUILDPYTHON) $(PROFILE_TASK) || true
-
-.PHONY: build_all_merge_profile
-build_all_merge_profile:
-	$(LLVM_PROF_MERGER)
```

Comment:
This change means that these targets can't be run manually.  Granted that's not something that's frequently done, but now it's impossible without painstakingly copying out what could be quite a long command.  I'm not sure this was worth doing, particularly since it doesn't even fully save us from recursive `make` calls.

Reply:
`profile-gen-stamp` saves us from a recursive `make` call; `profile-run-stamp` does not.

What's the use-case for running those targets manually? In my custom LLVM coverage script I generate and merge profile data in a single step, just like here. FWIW, I've never run them one at a time.

OTOH, I agree that the rationale for doing this particular refactor was a little weak; it would have been just fine to keep it as it was.

Reply:
If you look at this change in isolation, the rationale was a bit weak. But when you see the big picture (#104523), I think you'll see it is justified.

The current PGO code is a mess and I am going to incrementally clean it up.

Reply:
OK. Just don't be surprised by the bug reports ;) https://m.xkcd.com/1172/

Reply:
I used to be a maintainer of Firefox's build system. I'm very well versed in Hyrum's law :)

Hopefully within 1-2 weeks my refactor has finished merging and people will stop complaining about the build system getting confused when doing PGO. I fully intend to make `make` have a proper DAG for PGO builds. That means no more tribal incantations of make targets to work around gaps in the build system DAG.